evaluation/swe_bench/run_infer.py

```diff
@@ -383,10 +384,7 @@ def process_instance(
     if state is None:
         raise ValueError('State should not be None.')
 
-    # history is now available as a stream of events, rather than list of pairs of (Action, Observation)
-    # for compatibility with the existing output format, we can remake the pairs here
-    # remove when it becomes unnecessary
-    histories = state.history.compatibility_for_eval_history_pairs()
+    histories = [event_to_dict(event) for event in state.history.get_events()]
```

Comment:
Please note that this method filters out these types of events currently https://github.com/All-Hands-AI/OpenHands/blob/c0201d92f86133259c9bb685b360c14460b8ff02/openhands/memory/history.py#L31

Basically the reason for the existence of the compatibility function is to bring back NullActions and NullObservations when they're necessary for the expected format (action, obs) to be still respected.

Reply:
yeah i think this is actually good for eval purpose - only saving stuff that's needed?

Reply:
IMHO it's good, but it doesn't always match the pattern. Just wondering, will that work fine with the visualizer? Should it be done in all evals eventually?

Reply:
haven't fixed the visualizer for this yet - but eventually i'd actually hope to re-use the Share-OD visualizer for all eval